### PR TITLE
Improve PII Sanitizer, don't sanitize certain fields

### PR DESCRIPTION
**Pull request by saneshark**
## Description of change
ICN and EDIPI while PII, are just primary keys, and they're needed for correlating our logs with downstream partners. These fields cannot really be used to cause problems since you can't use an ICN or EDIPI to open a credit card for example. They're tremendously useful for debugging purposes.

## Testing done
specs

## Testing planned
none

#### Unique to this PR
Introduced an exceptions list. Since state gets triggered by regular expression, we will also check that whatever the key is, it is not in the exceptions list. There is a spec that tests that "state" is correctly filtered whereas relaystate, relayState, relay_state are not.

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by annaswims** [comment]
Can you connect this to an issue?

File: spec/lib/sentry/processor/pii_sanitizer_processor_spec.rb [hunk 1]
```diff
@@ -10,6 +10,35 @@
...
+        edipi: 'SOME EDIPI VALUE',
+        mhv_correlation_id: 'SOME MHV CORRELATION ID'
+      }
+    end
+
+    it 'sanitizes state' do
+      expect(result[:state]).to eq('FILTERED-CLIENTSIDE')
+    end
+
+    it 'does not saitize relay_state' do
```
Comment (lihanli) [inline spec/lib/sentry/processor/pii_sanitizer_processor_spec.rb]: typo

**Comment by saneshark** [comment]
I don't think there is any question that it is user identifying attribute, but I think a case can be made for the fact that it is needed for day to day operations, and my understanding is that that makes it ok. There isn't much one can do with an ICN or EDIPI, it's like having a primary key on a table in a database.

Also, we've been logging this for a very long time. Only stopped logging it sometime in the last two months. Bottom line, there is no way to really debug issues with downstream partners without these.

**Comment by annaswims** [comment]
Fair enough. I see that @lihanli was the one who added it, so since he approves I do too.

**Comment by saneshark** [comment]
Adding this for posterity:

From the [Eliminating the Unnecessary Collection and Use of Social Security Numbers at the Department of Veterans Affairs (rev 2014) p. 8](https://www.oprm.va.gov/docs/SSNPlan_2014.pdf) (pdf warning):

> The ICN is a sequentially assigned, non-intelligent number which, in itself,
> does not provide any identifying information about the patient. It conforms to the
> ASTM International standard for a universal healthcare identifier

**Pull request merged**